asyncio_mqtt/client.py

```diff
@@ -312,20 +327,37 @@ async def force_disconnect(self) -> None:
             self._disconnected.set_result(None)
 
     @_outgoing_call
-    async def subscribe(self, *args: Any, timeout: int = 10, **kwargs: Any) -> int:
-        result, mid = self._client.subscribe(*args, **kwargs)
+    async def subscribe(
+        self,
+        topic: Union[
+            str,
+            tuple[str, mqtt.SubscribeOptions],
+            list[tuple[str, mqtt.SubscribeOptions]],
+            list[tuple[str, int]],
+        ],
+        qos: int = 0,
+        options: Optional[mqtt.SubscribeOptions] = None,
+        properties: Optional[Properties] = None,
+        timeout: int = 10,
+    ) -> int:
+        result, mid = self._client.subscribe(topic, qos, options, properties)
         # Early out on error
         if result != mqtt.MQTT_ERR_SUCCESS:
             raise MqttCodeError(result, "Could not subscribe to topic")
         # Create future for when the on_subscribe callback is called
-        cb_result: "asyncio.Future[int]" = asyncio.Future()
+        cb_result = asyncio.Event()
         with self._pending_call(mid, cb_result, self._pending_subscribes):
             # Wait for cb_result
-            return await self._wait_for(cb_result, timeout=timeout)
+            return await self._wait_for(cb_result.wait(), timeout=timeout)
```

Comment:
The type annotation was wrong, the `Future` was of type `int` instead of the actual type `Union[Tuple[int], List[mqtt.ReasonCodes]]`.
So, the return should be of type `Union[Tuple[int], List[mqtt.ReasonCodes]]`.
It would be a breaking change to now really return an `int` instead of `Union[Tuple[int], List[mqtt.ReasonCodes]]`.